brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java

```diff
@@ -144,6 +146,20 @@ public void httpStatusCodeSettable_onUncaughtException_async() {
             "Can't currently control the HTTP status code on uncaught exception. #2656");
     }
 
+    @Test
+    @Override
+    public void httpRoute() throws IOException {
+        httpTracing = httpTracing.toBuilder().serverResponseParser(HttpResponseParser.DEFAULT::parse).build();
```

Comment:
here is where we are playing games to undo the change. having a consistent policy across all libraries is actually a goal of abstractions.

setting the span name to match the service name is very confusing, especially as a default. The motivation behind this could also drift as UI does.. for example, if this is done to effect a change in the UI it would have been better to request that change instead.

In general I think this change should be reverted as span name != service name and conflating them is very confusing. Please let me know if I somehow missed something thought. cc @anuraaga

Reply:
@adriancole Sorry for not telling you about this change. 😅 
I wanted to fill the gap between creating the decorator with `Tracing` and `HttpTracing`.
As you know, if the decorator is created using `Tracing` the span name is set as the service name.

And if you think we cannot set the service name as span name, I think we should also change the behavior in `ArmeriaHttp(Client|Server)Parser`. Any thoughts on this?

Reply:
I'm not offended, just could have helped reduce confusion. The `ArmeriaHttp(Client|Server)Parser` I thought of as "legacy" parsers for when people hadn't centralized brave config. If they are for longer term, yeah I would suggest re-thinking the default span name. Also, understanding what if any motivation there was for using service name as span name in case there's a UX issue we can help with

Reply:
So the default span name is the HTTP method when `HttpRequestParser.Default()` is used. Is that right?

Reply:
I think armeria often has access to the HTTP route information, so it should at least on servers be the full route

https://github.com/openzipkin/brave/tree/master/instrumentation/http#span-data-policy

Reply:
So should we set it back to `method path` and add an additional tag for the service name? 🤔 /cc @trustin

Reply:
path is never good because it can create high cardinality tags. it must be an http route really

Reply:
Yeah, what I meant was route. 😅

Reply:
Service name is generally presented in the Endpoint information. Perhaps you want to override the Endpoint service name with the Armeria one if a user didn't specify it to Brave?

Reply:
https://line-armeria.slack.com/archives/C1NGPBUH2/p1598171931036100
This is the related link where this PR came from.

One of our users wants to see the thrift service name from the UI.
Because we are setting the service name as the span name in `ArmeriaHttp(Client|Server)Parser`, the user could see it when he uses `BraveService.newDecorator(Tracing tracing)`.
But he couldn't when using `BraveService.newDecorator(HttpTracing httpTracing)`.

Reply:
Ah this reminds me of some conversations about it being a bit tricky instrumenting Armeria, which supports both HTTP and RPC depending on configuration. But I guess Armeria knows which one it is.

I think it's time to introduce `RpcTracing` support and make sure Armeria's defaults are different for HTTP and RPC.

https://github.com/openzipkin/brave/blob/master/instrumentation/rpc/src/main/java/brave/rpc/RpcTracing.java

Reply:
Thrift users I guess should be using `RpcTracing`

Reply:
What about for gRPC users? Do they need to use the `RpcTracing`?
I'm not sure if it's a good idea because we treat the Thrift and gRPC differently as we discussed here. 🤔 
https://github.com/line/armeria/issues/2480

Reply:
In terms of tracing, they should probably use `RpcTracing` as they are RPC and should have RPC tracing semantics - the trickiness in implementation I refer to indeed comes from gRPC being implemented as an HTTP service :P But I guess some special casing could work ok.

Reply:
> Some special casing

For example?